agent/plugins/modules/preset_config.py

```diff
@@ -0,0 +1,444 @@
+
+    ssh_pub_key = read_ssh_key_file(home_dir=home_dir)
+    pull_secret = read_pull_secret_file(working_dir=module.params['working_dir'])
+
+    facts_base = generate_preset_base(
+        home_dir=home_dir,
+        ssh_pub_key=ssh_pub_key,
+        pull_secret=pull_secret,
+        ci_token=module.params['ci_token'],
+        ci_server=module.params['ci_server'],
+        working_dir=module.params['working_dir'],
+        openshift_release_stream=module.params['openshift_release_stream'],
+        openshift_release_type=module.params['openshift_release_type'],
+        openshift_version=module.params['openshift_release_stream'],
+        cluster_name=module.params['cluster_name'],
+        base_domain=module.params['base_domain']
+    )
+
+    facts_cluster = determine_cluster_topology(
+        cluster_topology=module.params['cluster_topology'],
+        resource_profile=module.params['resource_profile'],
+        extra_workers_profile=module.params['extra_workers_profile']
+    )
+
+    facts_network = determine_network_topology(
+        ip_stack=module.params['ip_stack'],
+        host_ip_stack=module.params['host_ip_stack'],
+        provisioning_network_profile=module.params['provisioning_network_profile']
+    )
+
+    facts_agent = generate_preset_agent(
+        agent_static_ip_node0_only=module.params['agent_static_ip_node0_only']
+    )
+
+    facts = {**facts_base, **facts_cluster, **facts_network, **facts_agent}
+
+    # facts['node_hostnames'] = generate_node_hostnames_for_static_ips(
+    #     cluster_name=facts_base['cluster_name'],
+    #     num_masters=int(facts_cluster['num_masters']),
+    #     num_workers=int(facts_cluster['num_workers']),
+    #     agent_static_ip_node0_only=facts_agent['agent_static_ip_node0_only']
+    # )
+
+    if module.check_mode:
+        module.exit_json(**result)
+
+    result['ansible_facts']['devscripts'] = facts
+    module.exit_json(**result)
+
+def read_ssh_key_file(home_dir, file="/.ssh/id_rsa.pub", location=None):
+
+    if location:
+        ssh_pub_key_file = location
+    else:
+        ssh_pub_key_file = home_dir+file
+
+    with open(ssh_pub_key_file) as file:
+        ssh_pub_key = file.read()
+    file.close()
+
+    return ssh_pub_key.strip()
+
+def read_pull_secret_file(working_dir, file="/pull_secret.json", location=None):
+
+    if location:
+        pull_secret_file = location
+    else:
+        pull_secret_file = working_dir+file
+
+    with open(pull_secret_file) as file:
+        pull_secret = file.read()
+    file.close()
+
+    return pull_secret.strip()
+
+
+def generate_preset_base(home_dir, ci_token, ci_server, working_dir, ssh_pub_key, pull_secret,
+                        openshift_release_stream, openshift_release_type, openshift_version,
+                        cluster_name, base_domain):
+
+    cluster_domain = cluster_name+"."+base_domain
+    ocp_dir = working_dir+"/ocp/"+cluster_name
+    manifests_dir = ocp_dir+"/manifests"
```

Comment:
Is this supposed to point to the agent manifests? Because in such case it should be `cluster-manifets`

Reply:
We didn't make that change yet though. It's in #1406.

Reply:
So this is what is based off in the current file. But I dont think changing it now would break anything